table/block_based/block_based_table_builder.cc

```diff
@@ -1500,9 +1490,14 @@ void BlockBasedTableBuilder::BGWorkWriteMaybeCompressedBlock() {
       continue;
     }
 
+    NotifyCollectTableCollectorsOnBlockAdd(r->table_properties_collectors,
```

Comment:
It would be nice in PR summary to mention in one line that moving "NotifyCollectTableCollectorsOnBlockAdd()" out from "CompressAndVerifyBlock()" called in parallel to "BGWorkWriteMaybeCompressedBlock()" called sequentially is the fix. It was a bit lost in the explanation for the plumbing and refactoring in the PR summary.